tests/ignite/metrics/test_precision.py

```diff
@@ -47,168 +47,145 @@ def test_binary_wrong_inputs():
 
+    def _test(y_pred, y, batch_size):
         pr.reset()
-        y_pred = torch.randint(0, 2, size=(10,))
-        y = torch.randint(0, 2, size=(10,)).long()
         pr.update((y_pred, y))
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert pr._type == "binary"
-        assert isinstance(pr.compute(), float if average else torch.Tensor)
-        pr_compute = pr.compute() if average else pr.compute().numpy()
-        assert precision_score(np_y, np_y_pred, average="binary") == pytest.approx(pr_compute)
 
-        pr.reset()
-        y_pred = torch.Tensor([0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.51])
-        y_pred = torch.round(y_pred)
-        y = torch.randint(0, 2, size=(10,)).long()
-        pr.update((y_pred, y))
         np_y = y.numpy().ravel()
         np_y_pred = y_pred.numpy().ravel()
+
+        if batch_size > 1:
+            n_iters = y.shape[0] // batch_size + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                pr.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+
         assert pr._type == "binary"
         assert isinstance(pr.compute(), float if average else torch.Tensor)
         pr_compute = pr.compute() if average else pr.compute().numpy()
         assert precision_score(np_y, np_y_pred, average="binary") == pytest.approx(pr_compute)
 
-        # Batched Updates
-        pr.reset()
-        y_pred = torch.randint(0, 2, size=(100,))
-        y = torch.randint(0, 2, size=(100,)).long()
+    def get_test_cases():
 
-        batch_size = 16
-        n_iters = y.shape[0] // batch_size + 1
+        test_cases = [
+            (torch.randint(0, 2, size=(100,)), torch.randint(0, 2, size=(100,)), 1),
+            (torch.randint(0, 2, size=(100, 1)), torch.randint(0, 2, size=(100, 1)), 1),
+            (torch.randint(0, 2, size=(200,)), torch.randint(0, 2, size=(200,)), 1),
+            (torch.randint(0, 2, size=(200, 1)), torch.randint(0, 2, size=(200, 1)), 1),
+            # updated batches
+            (torch.randint(0, 2, size=(100,)), torch.randint(0, 2, size=(100,)), 16),
+            (torch.randint(0, 2, size=(100, 1)), torch.randint(0, 2, size=(100, 1)), 16),
+            (torch.randint(0, 2, size=(200,)), torch.randint(0, 2, size=(200,)), 16),
+            (torch.randint(0, 2, size=(200, 1)), torch.randint(0, 2, size=(200, 1)), 16),
+        ]
 
-        for i in range(n_iters):
-            idx = i * batch_size
-            pr.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+        return test_cases
 
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert pr._type == "binary"
-        assert isinstance(pr.compute(), float if average else torch.Tensor)
-        pr_compute = pr.compute() if average else pr.compute().numpy()
-        assert precision_score(np_y, np_y_pred, average="binary") == pytest.approx(pr_compute)
+    for _ in range(10):
```

Comment:
Let's keep 5 instead of 10 to reduce computational load.

Reply:
I thought about doing that, thanks!